tests/integrationtest/r/expression/json.result

```diff
@@ -771,3 +771,49 @@ insert into t values (NULL, NULL, NULL);
 select json_valid(j), json_valid(str), json_valid(other) from t;
 json_valid(j)	json_valid(str)	json_valid(other)
 NULL	NULL	NULL
+DROP TABLE IF EXISTS t1;
+CREATE TABLE t1(id INT PRIMARY KEY, d1 DATE, d2 DATETIME, t1 TIME, t2 TIMESTAMP, b1 BIT, b2 BINARY);
+INSERT INTO t1 VALUES (1, '2024-06-14', '2024-06-14 09:37:00', '09:37:00', '2024-06-14 09:37:00', b'0', 0x41);
+SELECT JSON_TYPE(d1) FROM t1;
+Error 3146 (22032): Invalid data type for JSON data in argument 1 to function json_type; a JSON string or JSON type is required.
+SELECT JSON_TYPE(d2) FROM t1;
+Error 3146 (22032): Invalid data type for JSON data in argument 1 to function json_type; a JSON string or JSON type is required.
+SELECT JSON_TYPE(t1) FROM t1;
+Error 3146 (22032): Invalid data type for JSON data in argument 1 to function json_type; a JSON string or JSON type is required.
+SELECT JSON_TYPE(t2) FROM t1;
+Error 3146 (22032): Invalid data type for JSON data in argument 1 to function json_type; a JSON string or JSON type is required.
+SELECT JSON_TYPE(b1) FROM t1;
+Error 3146 (22032): Invalid data type for JSON data in argument 1 to function json_type; a JSON string or JSON type is required.
+SELECT JSON_TYPE(b2) FROM t1;
+Error 3144 (HY000): Cannot create a JSON value from a string with CHARACTER SET 'binary'.
```

Comment:
The sql state seems not changed.

Reply:
Forgot to update the test result. It's changed now :+1: